app/src/main/java/de/mathfactory/mooltifill/MooltipassDevice.kt

```diff
@@ -428,18 +430,71 @@ class MooltipassDevice(private val device: BluetoothDevice, private var debug: I
 @ExperimentalCoroutinesApi
 class MooltipassScan {
 
+    private var context:Context? = null
```

Comment:
Please do not store the context when it is not necessary! It should preferably be passed and I think you could pass it easily to filter() from the pairedDevice(context: Context) function. Please note also my comment about storing the context already in #24.